modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaFieldWidgetTest.php

```diff
@@ -0,0 +1,226 @@
+        'handler_settings' => [
+          'target_bundles' => [
+            'image' => 'image',
+            'document' => 'document',
+          ],
+        ],
+        'sort' => [
+          'field' => '_none',
+        ],
+        'auto_create' => '0',
+      ],
+      'required' => FALSE,
+    ]);
+    $this->field->save();
+
+    // Setup the display options for form and view.
+    $form_display_options = [
+      'type' => 'oe_featured_media_autocomplete',
+      'settings' => [
+        'match_operator' => 'CONTAINS',
+        'match_limit' => 10,
+        'size' => 60,
+      ],
+    ];
+    $view_display_options = [
+      'type' => 'oe_featured_media_label',
+      'label' => 'above',
+      'settings' => [
+        'link' => TRUE,
+      ],
+    ];
+
+    // Prepare the default form display for rendering.
+    $display = \Drupal::service('entity_display.repository')
+      ->getFormDisplay($this->field->getTargetEntityTypeId(), $this->field->getTargetBundle())
+      ->setComponent($this->fieldStorage->getName(), $form_display_options);
+    $display->save();
+
+    // Prepare the default view display for rendering.
+    $display = \Drupal::service('entity_display.repository')
+      ->getViewDisplay($this->field->getTargetEntityTypeId(), $this->field->getTargetBundle())
+      ->setComponent($this->fieldStorage->getName(), $view_display_options);
+    $display->save();
+  }
+
+  /**
+   * Tests the featured media widget.
+   */
+  public function testFeaturedMediaWidget(): void {
+    // Visit the node add page.
+    $this->drupalGet('node/add/page');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    // Assert that the fields and labels are present on the page.
+    $assert_session->pageTextContains('Featured media');
+    $assert_session->fieldExists('Media item');
+    $assert_session->fieldExists('Caption');
+
+    // Assert the help texts without media overview permission.
+    $assert_session->pageTextContains('Start typing the name of the Media.');
+    $assert_session->pageTextNotContains('You can manage all the media items on this page.');
+    $assert_session->pageTextContains('Allowed media types: Image, Document');
+
+    // Login with a user with extended permissions.
+    $this->drupalLogin($this->drupalCreateUser([
+      'access content',
+      'access media overview',
+      'create image media',
+      'create page content',
+    ]));
+    $this->drupalGet('node/add/page');
+
+    // Assert the help texts with media overview permission.
+    $assert_session->pageTextContains('Start typing the name of the Media.');
+    $assert_session->pageTextContains('You can manage all the media items on this page.');
+    $assert_session->pageTextContains('Allowed media types: Image, Document');
+
+    // Test that the Media item field turns required once the Caption is filled.
+    $this->assertFalse($page->findField('Media item')->hasAttribute('required'));
```

Comment:
Let's not mix usages of assert_session and page. The method `$assert_session->fieldExists()` returns the dom element, let's stick to assert_session.